pkgs/development/compilers/llvm/12/default.nix

```diff
@@ -17,6 +19,7 @@
     then null
     else pkgs.bintools
 , darwin
+, ... # Allows passthrough to packages via newScope.
```

Comment:
> OK. Are you happy if I add , ... to this patch? That way it would be possible to override other things in the future should they be added, and the top level documents the arguments to all llvmPackages as they are today, at least.

(i didn't read the last part of this comment correctly when responding to it).

thanks for adding the `...`.

> the top level documents the arguments to all llvmPackages as they are today, at least.

so i am not sure about having the union of all llvmPackages subpackage arguments in the lop level given they can go stale and it is somewhat non-obvious what all the extra arguments are for given they aren't used in `llvm/[:digit:]/default.nix`. my preference would be to only have the args used in `llvm/[:digit:]/default.nix` and let the `...` take care of the rest.

Reply:
tho -- maybe i'm missing something as it seems that before your change there are a bunch of extra args that aren't used in the top level `default.nix`, for example `libxml2`

Reply:
I have modified the PR and updated the title and commit message accordingly. I have also gone to the trouble of removing anything which was not referenced in the default.nix.